ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/SOLR/package/scripts/params.py

```diff
@@ -84,6 +92,10 @@ def get_name_from_principal(principal):
 # Only supporting SolrCloud mode - so hardcode those options
 solr_cloudmode = 'true'
 solr_dir = '/usr/lib/solr'
+
+if stack_version_formatted and check_stack_feature(StackFeature.ROLLING_UPGRADE, stack_version_formatted):
+  solr_dir = format("{stack_root}/current/solr-server")
```

Comment:
Can we use `component_directory` here? Like [Zookeeper](https://github.com/apache/ambari/blob/trunk/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/ZOOKEEPER/package/scripts/params_linux.py#L55)?

Reply:
That's a good reminder, but I found `stack_packages.json` has the map relation, should we obtain it from here?
e.g.
![1666225375739](https://user-images.githubusercontent.com/16263438/196828556-1973491c-6c00-4f77-81c2-b391bf6859f9.png)